### PR TITLE
Insert [filter weapon] in filter student/opponent/etc.] for weapon like abilities(1.15/1.16)

### PR DESCRIPTION
**Pull request by newfrenchy83**
i doing stupid thing in old PR and a post another in replacement for resolve https://github.com/wesnoth/wesnoth/issues/4389

### CODE REVIEW

**Comment by newfrenchy83** [comment]
@CelticMinstrel , @gfgtdf Forgive me for insisting but could someone look into this PR, please?

**Comment by jostephd** [comment]
@CelticMinstrel @newfrenchy83 Is this a 1.16 blocker? Does it fix a regression, or just add a new feature?

**Comment by newfrenchy83** [comment]
Is adda new feature for what abilities like weapon was similar to true weaoon special in wml code

**Comment by CelticMinstrel** [comment]
@jostephd From my understanding it's a modification of syntax for greater consistency.

@newfrenchy83 Does this actually remove `[filter_weapon]` and `[filter_second_weapon]`? In case it does, we can't just remove them. They should produce a deprecation warning when used, but they should still continue to work. As long as that's the case, this can be merged.

**Comment by newfrenchy83** [comment]
@CelticMinstrel i fear to let [filter_weapon] and [filter_second_weapon] because possibity of bugs if someone put in abilities weapon [filter_weapon] int [filter_student/opponent/etc) and filter_weapon/filter_second_weapon] simultaneous. and i don't see where put deprecation message in that case.

**Comment by newfrenchy83** [comment]
[filter_weapon/filter_second_weapon] not deprecated for [leadership] and [resistance] abilities, i can't put deprecated_message in ability_affects_weapon

**Comment by newfrenchy83** [comment]
an idea will be to using weapon for [leadership] and to erase [filter_second_weapon] and filter_weapon to [resistance]

**Comment by CelticMinstrel** [comment]
If they use both `[filter_(second_)weapon]` and the nested `[filter_weapon]` that should replace it, then it's fair to make the new syntax take priority over the old - so just ignore the deprecated one (other than printing a deprecation message).

**Comment by newfrenchy83** [comment]
> 
> 
> If they use both `[filter_(second_)weapon]` and the nested `[filter_weapon]` that should replace it, then it's fair to make the new syntax take priority over the old - so just ignore the deprecated one (other than printing a deprecation message).

and how i doing that? It willbe simpler to 

> 
> 
> If they use both `[filter_(second_)weapon]` and the nested `[filter_weapon]` that should replace it, then it's fair to make the new syntax take priority over the old - so just ignore the deprecated one (other than printing a deprecation message).

and how i doing that please?

File: src/actions/attack.cpp [hunk 1]
```diff
@@ -1687,7 +1691,7 @@ std::pair<int, bool> ability_leadership(const std::string& ability,const unit_ma
 		} else if(up == units.end() && (filter_student || filter || filter_attacker || filter_defender)) {
 			return {abil_value, false};
 		} else {
-			show_result = !(!un->abilities_filter_matches(*i->first, attacker, abil_value) || ability_apply_filter(un, up, ability, *i->first, loc, opp_loc, attacker));
+			show_result = !(!un->abilities_filter_matches(*i->first, attacker, abil_value) || ability_apply_filter(un, up, ability, *i->first, loc, opp_loc, attacker, weapon, opp_weapon));
```
Comment (newfrenchy83) [inline src/actions/attack.cpp]: ability_apply_filter in after un->get_abilities(ability, weapon, opp_weapon); [filter_weapon/filtr_second_weapon] will be used in all case or these filter will be in abilitie like weapon in same time what usual code syntax what i want emulate, impossible to create an priority to the last, the only solution is to erase [filter_weapon/second_weapon] directly. It is one or other but not both.

**Comment by newfrenchy83** [comment]
> 
> 
> If they use both `[filter_(second_)weapon]` and the nested `[filter_weapon]` that should replace it, then it's fair to make the new syntax take priority over the old - so just ignore the deprecated one (other than printing a deprecation message).

@CelticMinstrel i create priority for new syntax, i new syntax is present, old willbe blocked

**Pull request closed without merging**